### ISSUE TITLE
User Management phase II: persistent gameplay identity + game history

### ISSUE THREAD

**Issue by jdarthur** [jdarthur/trivia#11]
## Motivation

Today a player gets a freshly generated, ephemeral UUID every time they join a
game. Nothing connects the person who played last Thursday to the person who
plays this Thursday, so there is no way to look back at games you played, and
every join means retyping your team name and re-picking your icon.

The read side of this is now much cheaper than when this ticket was filed in
2021: #234 added the per-question cumulative score-history API and #237/#248
added the graph modal and PNG export. `getSessionScoreHistory` builds its series
purely from stored `session_question` / `answer` rows with no dependence on live
session state, so it already works against a finished game. The remaining work
is mostly *linking* a durable account to the per-game `player` rows and adding
read surfaces on top.

## Original sketch (2021)

> - Play as logged-in or guest user
> - if logged-in, create a player record with ones `userId` attached
> - save the players `realName` and last-used icon
>
> ### User browser
> - Go to a logged-in player's page to see summary, games played
> - Click on a game played to see podium, scores graph, etc.

That design still holds. The notes below fill it in against the code as it
exists now.

### Identity is not auth — keep these separate

- **#256's per-player token** is a per-session *capability*: "this browser is
  this player in this game." Ephemeral, dies with the session.
- **This ticket's account** is durable *identity*: the thing history hangs off.
  It is a feature, not a security control.

Persistent identity does **not** replace #256, because guest play has to
survive. The app's whole distribution model is an invite link pasted into a
group chat; requiring a login to answer trivia questions would be a product
regression. So there will always be unauthenticated players, and they still
need per-session authorization. Auth0 on the gameplay side is strictly
*additive*.

**Consequence for sequencing: #256 lands first.** It is the security fix, it is
smaller, and `CreatePlayer` / `CreateSession` are touched by both — doing them
in the other order means editing the same functions twice.

### Auth0, not a localStorage credential

An unrecoverable user ID in `localStorage` is not identity, it is a permanent
bearer token — #256's problem with the expiry removed and the stakes raised,
since it would guard a year of history instead of one evening's game. Its
failure modes (cleared site data, new device, new browser, Safari's eviction of
script-writable storage) all land precisely at *"look back at it later"*, which
is the entire value proposition here.

The rule of thumb applied throughout this ticket: **`localStorage` as a cache is
fine, `localStorage` as the system of record is not.** Hence #258 (guest
preferences, losing them costs a retype) uses it happily, and identity does not.

### Don't replace the ephemeral player ID — link to it

A `player` row per game is *correct*. It is the per-game participation record,
and `answer`, `session_score`, `session_player`, and reactions all key off
`player.id` per session. Adding a fixed user ID in its place would mean
rewriting all of that.

Instead add a nullable link upward:

```sql
ALTER TABLE player ADD COLUMN user_id TEXT NOT NULL DEFAULT ''  -- '' = guest
```

Guests leave it empty and behave exactly as today. Logged-in players get it
stamped at join. History then becomes a join over data that already exists:
`player` → `session_player` → `session`, filtered on `player.user_id`.

### 1. The `user` table is currently dev-mode-only scaffolding

`user` (migration 7: `sub`, `username`, `display_name`, `create_date`,
`avatar_url`) is **never written or read in production**:

- The only `INSERT INTO user` in the tree is `store.SeedUsers`
  (`store/seed.go:28`), called solely under `--dev-mode` (`main.go:70`).
- The only `SELECT ... FROM user` is in `decodeDevToken`
  (`common/common_middleware.go:241`), also dev-mode only. The production path,
  `DecodeToken`, validates the JWT and reads `sub` from the claims without ever
  touching the database.
- Every `user_id` column (`question`, `round`, `game`, `category`,
  `collection`, `scoring_note`) is a bare `TEXT NOT NULL DEFAULT ''` with **no
  `REFERENCES user(sub)`**.

So there is no user provisioning anywhere. That is fine for the editor, which
only ever needs `sub` as an opaque scoping key — but a "user browser" page has
to render a display name and avatar, which means provisioning becomes a real
prerequisite (#260). Nobody would guess this from the ticket.

### 2. Deleting a session destroys every participant's history of that game

`DeleteSession` (`sessions/session.go:125`) is a mod-only hard delete, and
everything hangs off `session(id)` with `ON DELETE CASCADE`: `session_player`,
`session_question`, `session_score`, `answer`, `session_state`, and all the
`session_question_*` tables (migration 1, lines 144-190 and following).

`session_player` and `session_score` *also* cascade from
`player_id REFERENCES player(id) ON DELETE CASCADE`, so #254's
`DELETE /gameplay/player/:id` likewise wipes that player's scores and
memberships.

Right now that is invisible, because nothing reads a session after it ends. The
moment history exists, one person deleting "their" session silently deletes
everyone else's record of that night. This needs an explicit decision before the
read side is built, because it determines whether the history API reads a live
join or a snapshot — see **Open decisions**.

## Sub-tickets

Grouped by phase; each is independently landable.

**Phase 0 — no accounts needed, ships immediately**
- #258 — Guest team name + icon memory in `localStorage`

**Phase 1 — identity infrastructure** (depends on #256)
- #259 — Gameplay Auth0 audience + multi-audience token verification
- #260 — User provisioning: upsert the `user` row on login
- #261 — Schema + write path: `player.user_id` stamped at join
- #262 — Optional login in the gameplay lobby (guest stays the default)

**Phase 2 — carry-forward**
- #263 — Server-side profile carry-forward for logged-in players

**Phase 3 — history**
- #264 — History durability: decide and implement session-delete behavior
- #265 — History API: a user's played games
- #266 — History list page ("user browser")
- #267 — Single-game detail view: podium + score graph
- #268 — Aggregate summary stats

**Phase 4 — tests**
- #269 — Test coverage: guest path unchanged, logged-in path, history

## Out of scope

- Replacing the per-game `player` row with a global user identity (see Design).
- Making login *required* for gameplay. Guest play is a hard requirement.
- Cross-user social features (friends, leaderboards across sessions, rivalries).
- Merging a guest's past games into an account retroactively — see
  **Not backfillable** below.
- Editor-side user management (#10, done).

### Not backfillable

Existing `player` rows carry no information about who the human was, so there is
no way to attribute historical games to a newly created account. Everyone's
history starts empty at the point #261 ships. Don't spend time on a backfill;
just don't let it come as a surprise.

## Open decisions

- **Auth0 audience.** `common/common_middleware.go:47` hardcodes a single
  audience (`https://borttrivia.com/editor`) and `AsUser` accepts any valid
  token for it. If gameplay logins reuse that audience, every trivia player
  holds a token that passes `AsUser`. That is *not* a data-exposure bug —
  editor content is properly per-user scoped (`question.user_id` etc., and the
  handlers filter on `GetUserId`), so a new signup sees their own empty
  workspace. But it does mean "anyone who logs in to play trivia silently
  becomes a question author."
  *(Recommended)* a separate gameplay audience, so the two surfaces stay
  distinguishable and a gameplay token cannot reach `/editor/*` at all.
  *(Alternative)* reuse the editor audience and accept the overlap — cheaper,
  and arguably a feature if you'd like players to build their own games.
  **This is the one decision to settle before #259 starts.** MAU cost is not
  a factor (private site, friends only).

- **History durability** (see finding 2). *(Recommended, cheap)* leave the
  cascade alone, add a confirmation dialog to `DeleteSession` that says history
  will be lost, and document that finished sessions are not meant to be
  deleted. *(Alternative, correct)* soft-delete sessions (`deleted_at`) and/or
  snapshot a immutable per-game summary row at scoring completion, so history
  survives deletion. The second is real work and probably not warranted for a
  private site — but it has to be chosen deliberately, because #265 reads
  whichever model wins.

- **History visibility.** Own games only, or can you see another player's
  profile page? *(Recommended)* own history only to start; a session's
  scoreboard is already visible to its participants, which covers the social
  case without a new disclosure surface.

## Acceptance criteria

- [ ] Guests can still join a game from an invite link with zero extra clicks
      and no login (non-negotiable regression guard)
- [ ] Guests get their team name and last-used icon prefilled on a later visit
- [ ] A player can optionally sign in on the gameplay side; a signed-in player's
      `player` row carries their `user_id`
- [ ] Signed-in players get their team name and icon carried forward server-side
- [ ] A signed-in user has a page listing the games they have played
- [ ] Clicking a played game shows its podium and score graph
- [ ] Summary stats (games played, wins, average placement) render on the profile
- [ ] A gameplay token cannot author editor content, or the overlap is a
      documented deliberate choice
- [ ] Session-delete behavior with respect to history is documented and
      intentional
- [ ] `README.md` / `AGENTS.md` describe gameplay identity as optional and
      additive to #256's per-session capability
